doc: update installation documentation

lukpueh opened the pull request:
Fixes #1846 

**Description of the changes being introduced by the pull request**:
Update severely outdated installation documentation.

- Simplify "Simple Installation" section
- Update "Release Verification" section to actually verify a tuf release and with a key of an active maintainer
- Update and simplify section about non-python dependencies (just point to installation instructions for underlying crypto backends, they are up-to-date and have become a lot easier)
- Add "Development installation" section

This PR also updates the install section in the contributing doc, plus other small fixes. See commits for details!


**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

Commit:
doc: update installation documentation

Update severely outdated installation documentation.

- Simplify "Simple Installation" section
- Update "Release Verification" section to actually verify a tuf
  release and with a key of an active maintainer
- Update and simplify section about non-python dependencies
  (just point to installation instructions for underlying crypto
  backends, they are up-to-date and have become a lot easier)
- Add "Development installation" section

Signed-off-by: Lukas Puehringer <lukas.puehringer@nyu.edu>

Commit:
doc: update install section in contributing doc

Replace custom installation section in contribution docs with
pointer to updated installation documentation.

Also configure sphinx autosectionlabel for cross-document refs.

Signed-off-by: Lukas Puehringer <lukas.puehringer@nyu.edu>

Commit:
doc: minor title changes for rtd navbar

Make contributing document header sentence case for consistency
with other docs and shorten menu name in side navbar to stand out
less.

Signed-off-by: Lukas Puehringer <lukas.puehringer@nyu.edu>

Commit:
doc: fix link to tox docs

Signed-off-by: Lukas Puehringer <lukas.puehringer@nyu.edu>

lukpueh commented on docs/INSTALLATION.rst:
```diff
@@ -1,93 +1,86 @@
-   gpg: Good signature from "Vladimir Diaz (Vlad) <vladimir.v.diaz@gmail.com>" [ultimate]
 
+Verify release signatures
+-------------------------
```
I find this section sticking out a bit too much for installation instructions. Should we shorten it, or at least move it to the bottom of the document?

jku replied:
I think I agree. I'd like to promote signature verification but when the process looks like this (even while handwaving the PGP key lookup part)... I think it breaks the flow of the document and hides the other install options below the fold.

Considering we have three different install methods (normal, native crypto, dev)... maybe list those three first and then verification as the last thing?

lukpueh replied:
see 0216740

lukpueh commented:
See https://theupdateframework--1853.org.readthedocs.build/en/1853/INSTALLATION.html for how the docs look on RTD.

lukpueh commented on docs/CONTRIBUTING.rst:
```diff
@@ -9,37 +9,14 @@ All submitted code should follow our `style guidelines
-    $ python3 -m pip install -r requirements-dev.txt
+.. note::
 
+     Also see :ref:`development installation instructions <installation:Install for development>`.
```
This reference is not resolved when the rendered document is browsed on GitHub, which is likely to happen because it is the community standard location for a GitHub repo's contributing docs.

A quick fix would be to use the absolute RTD link (see https://github.com/theupdateframework/python-tuf/pull/1781#discussion_r791639408).

In the future we might want to think about separating docs for RTD and docs for GitHub (see #1849)

jku replied:
yeah, it does look like sharing docs is tricky and results in issues like this.

lukpueh replied:
see cd70e47

jku reviewed (approved):
Nice improvement. I agree with you on the verification comment though, won't object to moving that part further down.

Commit:
doc: move verify section in install docs

Move release signature verification instructions to bottom of
install docs. The doc is short, so the section is still prominent
enough for promoting verification, but does not break the reading
flow as much anymore.

Signed-off-by: Lukas Puehringer <lukas.puehringer@nyu.edu>

Commit:
doc: cross-doc absolute link hack

This is an ugly hack to also resolve the link when the document is
rendered in GitHub, where it is likely to be browsed, because it is
the community standard location for a GitHub repo's contributing
docs.

Coordinate with #1849 to better separate RTD docs with GitHub docs
in the future.

Signed-off-by: Lukas Puehringer <lukas.puehringer@nyu.edu>

lukpueh commented:
@jku, want to take another look at my latest changes? Or should we just merge?
